devices/sam/samd21.xml

```diff
@@ -0,0 +1,704 @@
+    <valid-device>samd21g18a</valid-device>
+    <valid-device>samd21g18au</valid-device>
+    <valid-device>samd21j15a</valid-device>
+    <valid-device>samd21j15b</valid-device>
+    <valid-device>samd21j16a</valid-device>
+    <valid-device>samd21j16b</valid-device>
+    <valid-device>samd21j17a</valid-device>
+    <valid-device>samd21j17d</valid-device>
+    <valid-device>samd21j18a</valid-device>
+    <driver name="core" type="m0plus">
+      <vector position="-15" name="Reset"/>
+      <vector position="-14" name="NonMaskableInt"/>
+      <vector position="-13" name="HardFault"/>
+      <vector position="-5" name="SVCall"/>
+      <vector position="-2" name="PendSV"/>
+      <vector position="-1" name="SysTick"/>
+      <vector position="0" name="PM"/>
+      <vector position="1" name="SYSCTRL"/>
+      <vector position="2" name="WDT"/>
+      <vector position="3" name="RTC"/>
+      <vector position="4" name="EIC"/>
+      <vector position="5" name="NVMCTRL"/>
+      <vector position="6" name="DMAC"/>
+      <vector device-variant="a|au|b|bu|cu|d|du" position="7" name="USB"/>
+      <vector position="8" name="EVSYS"/>
+      <vector position="9" name="SERCOM0"/>
+      <vector position="10" name="SERCOM1"/>
+      <vector position="11" name="SERCOM2"/>
+      <vector position="12" name="SERCOM3"/>
+      <vector device-pin="g|j" position="13" name="SERCOM4"/>
+      <vector device-pin="g|j" position="14" name="SERCOM5"/>
+      <vector position="15" name="TCC0"/>
+      <vector position="16" name="TCC1"/>
+      <vector position="17" name="TCC2"/>
+      <vector position="18" name="TC3"/>
+      <vector position="19" name="TC4"/>
+      <vector position="20" name="TC5"/>
+      <vector device-pin="g" device-variant="au|l" position="21" name="TC6"/>
+      <vector device-pin="j" device-variant="a|b|d" position="21" name="TC6"/>
+      <vector device-pin="g" device-variant="au|l" position="22" name="TC7"/>
+      <vector device-pin="j" device-variant="a|b|d" position="22" name="TC7"/>
+      <vector position="23" name="ADC"/>
+      <vector position="24" name="AC"/>
+      <vector position="25" name="DAC"/>
+      <vector device-variant="a|au|b|bu|cu|d|du" position="26" name="PTC"/>
+      <vector device-variant="a|au|b|bu|cu|d|du" position="27" name="I2S"/>
+      <vector device-variant="l" position="28" name="AC1"/>
+      <vector device-flash="17" device-variant="d|du|l" position="29" name="TCC3"/>
+      <memory device-flash="15" device-variant="b|bu|cu|l" name="eeprom" size="1024"/>
+      <memory device-flash="16" device-variant="b|bu|cu|l" name="eeprom" size="2048"/>
+      <memory device-flash="17" device-variant="d|du|l" name="eeprom" size="4096"/>
+      <memory device-flash="15" name="flash" size="32768"/>
+      <memory device-flash="16" name="flash" size="65536"/>
+      <memory device-flash="17" name="flash" size="131072"/>
+      <memory device-flash="18" name="flash" size="262144"/>
```

Comment:
No SRAM?

Reply:
What about start addresses? You'll need that later for the [linkerscript generator in `modm:platform:cortex-m`](https://github.com/modm-io/modm/blob/develop/src/modm/platform/core/cortex/linker.macros#L18-L25). They differ for device variant, due to RWW.

Reply:
everything has a different name... I'm sure this will be fun when I add more variants. done.